File: app/src/main/java/com/bhanit/tapthegrey/SplashActivity.java
package com.bhanit.tapthegrey;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.bhanit.tapthegrey.helper.Log;
import com.bhanit.tapthegrey.utils.constants.TapTheGrey;

public class SplashActivity extends AppCompatActivity {
    private static final String TAG = SplashActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        waitAndStartTapTheGreyActivity();
    }

    private void waitAndStartTapTheGreyActivity() {
        Log.d(TAG, "waitAndStartTapTheGreyActivity: ");
        Runnable runnable = this::startDashboardActivity;
        new Handler().postDelayed(runnable, TapTheGrey.Time.FIVE_SECOND);
    }

    private void startDashboardActivity() {
        Log.d(TAG, "startDashboardActivity() method called");
        Intent intent = new Intent(this, TapTheGreyActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/bhanit/games/tapthegrey/activity/SplashActivity.kt
package com.bhanit.games.tapthegrey.activity

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.WindowInsets
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import com.bhanit.games.tapthegrey.R
import com.bhanit.games.tapthegrey.helper.Log
import com.bhanit.games.tapthegrey.utils.constants.TapTheGrey

@SuppressLint("CustomSplashScreen")
class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        removeStatusBar()
        waitAndStartTapTheGreyActivity()
    }

    private fun removeStatusBar() {
        Log.d(TAG, "removeStatusBar: ")
        @Suppress("DEPRECATION")
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window?.insetsController?.hide(WindowInsets.Type.statusBars())
        } else {
            window.setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN
            )
        }
    }

    private fun waitAndStartTapTheGreyActivity() {
        Log.d(TAG, "waitAndStartTapTheGreyActivity: ")
        val runnable = Runnable { this.startDashboardActivity() }
        Handler(Looper.getMainLooper()).postDelayed(runnable, TapTheGrey.Time.ONE_SECOND.toLong())
    }

    private fun startDashboardActivity() {
        Log.d(TAG, "startDashboardActivity() method called")
        val intent = Intent(this, TapTheGreyActivity::class.java)
        startActivity(intent)
        finish()
    }

    companion object {
        private val TAG: String = SplashActivity::class.java.simpleName
    }
}